subsys/net/lib/nrf_cloud/src/nrf_cloud_pgps.c

```diff
@@ -143,7 +143,7 @@ static struct nrf_cloud_pgps_prediction *get_cached_prediction(off_t off)
 	if (prediction_cache_flash_offset != off) {
 		int err;
 
-		err = flash_area_read(prediction_flash_area, off,
+		err = flash_area_read(prediction_flash_area, off - prediction_flash_area->fa_off,
```

Comment:
Would it make more sense to offset write to be inside the prediction_flash_area, rather than offset the flash_area_read to be outside of prediction_flash_area?

It feels weird to me that the flash_area API gives us partition offsets, but then we write/read from the beginning of the device anyways, but it is also likely that that is intentional and I am just misunderstanding something

Reply:
I wrote this code originally with only stream_flash being used for access to flash within the application image in the SoC flash.  Reading was done directly from SoC flash, so it made more sense to store pointers for each prediction, rather than offsets into a partition.  In fact, I didn't even use a P-GPS-specific partition then, but overloaded the secondary mcuboot partition as an MVP compromise.

Reply:
Ahh, got it -- per our private conversation, `off` meant to be in flash device address space, not partition address space, and I had subconsciously assumed it was the latter rather than the former



I think a comment right above the flash_area_read along the lines of:

`Subtract fa_off to convert from flash device address space to partition address space.`

would be a good idea.

I also think, in addition to that, adding a doxygen comment for get_cached_prediction with a parameter description stating that the `off` arg is the offset in flash device address space would also help